remotes/docker/httpreadseeker.go

```diff
@@ -77,6 +77,15 @@ func (hrs *httpReadSeeker) Read(p []byte) (n int, err error) {
 			return n, nil
 		}
 	}
+	// close the connection if there is no more available data
```

Comment:
I think we should have a note about the reason why we need immediately release the resource (i.e. caller can rely on the status of the reader for monitoring purpose, etc.).

I'm also a bit curious about is it possible to run `src.Close()` before `content.Commit(dst)` to make sure releasing resources before the long task.

```
defer src.Close()
content.Commit(dst)
```

Reply:
> I think we should have a note about the reason why we need to immediately release the resource (i.e. caller can rely on the status of the reader for monitoring purpose, etc.).

Thanks! Updated. 

> I'm also a bit curious about whether it's possible to run src.Close() before content.Commit(dst) to make sure releasing resources before the long task.

I was thinking about this. But the `httpReadSeeker` can reconnect if there is unexpected EOF. And the pusher has `ErrReset` case and the content.Copy helper supports the retry. So, we can't just close the src here. 

https://github.com/containerd/containerd/blob/7deb68fbf4b550d089488840d5ba15c7e345b6db/remotes/docker/pusher_test.go#L164

https://github.com/containerd/containerd/blob/7deb68fbf4b550d089488840d5ba15c7e345b6db/content/helpers.go#L183